src/Aspire.Hosting.RabbitMQ/RabbitMQBuilderExtensions.cs

```diff
@@ -30,14 +34,26 @@ public static class RabbitMQBuilderExtensions
         var passwordParameter = password?.Resource ?? ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(builder, $"{name}-password", special: false);
 
         var rabbitMq = new RabbitMQServerResource(name, userName?.Resource, passwordParameter);
-        return builder.AddResource(rabbitMq)
-                      .WithEndpoint(hostPort: port, containerPort: 5672, name: RabbitMQServerResource.PrimaryEndpointName)
-                      .WithImage("rabbitmq", "3")
-                      .WithEnvironment(context =>
-                      {
-                          context.EnvironmentVariables["RABBITMQ_DEFAULT_USER"] = rabbitMq.UserNameReference;
-                          context.EnvironmentVariables["RABBITMQ_DEFAULT_PASS"] = rabbitMq.PasswordParameter;
-                      });
+        var rabbitmq = builder.AddResource(rabbitMq)
+                              .WithEndpoint(hostPort: port, containerPort: 5672, name: RabbitMQServerResource.PrimaryEndpointName)
+                              .WithEnvironment(context =>
+                              {
+                                  context.EnvironmentVariables["RABBITMQ_DEFAULT_USER"] = rabbitMq.UserNameReference;
+                                  context.EnvironmentVariables["RABBITMQ_DEFAULT_PASS"] = rabbitMq.PasswordParameter;
+                              });
+
+        if (builder.ExecutionContext.IsRunMode)
```

Comment:
It does feel like this is a diff approach than other local management experiences and implicit instead of explicit (even in run mode). e.g. why would we just add pgadmin when in runmode then? Even though it isn't a separate management container, I feel like we should model it as such.

Reply:
Because it's in the same container it makes composition of the various container extension methods much harder to reason about. When it's a separate container instance we don't have to worry so much about whether the tag or whole image has been changed on the resource being extended. If we wanted to model this as a separate method we'd need to think about how we handle situations where the main resource's image or tag aren't the default.

Reply:
Spoke to @davidfowl about this some more and I'm going to make the change to put this in a separate method and handle the known tag formats. If you call it and the container image registry, name or tag is of an unsupported value it'll throw.